experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -2316,6 +2329,80 @@ def _validate_primary_secondary_outcomes(self, data):
 
         return data
 
+    def _validate_firefox_labs(self, data):
+        if not data.get("is_firefox_labs_opt_in"):
+            return data
+
+        min_version = NimbusExperiment.Version.parse(data.get("firefox_min_version"))
+        required_min_version = NimbusExperiment.FIREFOX_LABS_MIN_VERSION.get(
+            self.instance.application
+        )
+
+        if required_min_version is None:
+            raise serializers.ValidationError(
+                {
+                    "is_firefox_labs_opt_in": NimbusExperiment.ERROR_FIREFOX_LABS_UNSUPPORTED_APPLICATION,
+                }
+            )
+        else:
+            required_min_version = NimbusExperiment.Version.parse(required_min_version)
+            if min_version < required_min_version:
+                raise serializers.ValidationError(
+                    {
+                        "firefox_min_version": NimbusExperiment.ERROR_FIREFOX_LABS_MIN_VERSION.format(
+                            version=required_min_version
+                        ),
+                    }
+                )
+
+        if not data.get("is_rollout"):
+            raise serializers.ValidationError(
+                {"is_rollout": NimbusExperiment.ERROR_FIREFOX_LABS_ROLLOUT_REQUIRED}
+            )
+
+        errors = {
+            field: [NimbusExperiment.ERROR_FIREFOX_LABS_REQUIRED_FIELD]
+            for field in (
+                "firefox_labs_title",
+                "firefox_labs_description",
+                "firefox_labs_group",
+            )
+            if not len((data.get(field) or "").strip())
+        }
+
+        if description_links := (
+            data.get("firefox_labs_description_links") or ""
+        ).strip():
+            try:
+                description_links_obj = json.loads(description_links)
+
+                if isinstance(description_links_obj, dict):
+                    if not all(
+                        is_valid_http_url(value)
+                        for value in description_links_obj.values()
+                    ):
+                        errors["firefox_labs_description_links"] = [
+                            NimbusExperiment.ERROR_FIREFOX_LABS_DESCRIPTION_LINKS_HTTP_URLS,
+                        ]
+
+                elif (
+                    not isinstance(description_links_obj, dict)
+                    and description_links_obj is not None
+                ):
+                    errors["firefox_labs_description_links"] = [
+                        NimbusExperiment.ERROR_FIREFOX_LABS_DESCRIPTION_LINKS_JSON,
+                    ]
+
+            except Exception:
```

Comment:
Maybe just JSONException here instead of all exceptions? 🤔

Reply:
Ive restructured this so only json.loads is in the try.